src/plugins/vis_types/timeseries/public/convert_to_lens/index.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Panel } from '../../common/types';
+import { PANEL_TYPES } from '../../common/enums';
+import { ConvertTsvbToLensVisualization } from './types';
+
+const getConvertFnByType = async (
```

Comment:
No need to delcare it as `async` as the only `await` call can be removed

Reply:
As you can see we have import inside inner function because of this that function is async

<img width="798" alt="Снимок экрана 2022-07-28 в 16 17 10" src="https://user-images.githubusercontent.com/16915480/181514571-09e115a7-9880-4b25-928d-24ea5f745b29.png">

Reply:
The `convertionFns[type]()` function returns a `Promise`, which is `await`-ed to have a `ConvertTsvb...` object, and then wrapped it again in a `Promise` (because the outer function is also `async`).
Removing just the `await` will make the result become a `Promise<Promise<ConvertTsvb...>>` which in turns gets flatten into a `Promise<ConvertTsvb...>`. Without the `await` there's no more need for the outer function to be `async` as well, returning the `convertionFns[type]()` `Promise` straight away.

Reply:
You miss the following: convertionFns[type] is async function (import inside) which return another async function (convertToLens) because of that we should do await for convertionFns[type]?.() so that after that can call convertToLens.